tests/components/wsdot/test_sensor.py

```diff
@@ -16,56 +13,46 @@
     RESOURCE,
     SCAN_INTERVAL,
 )
-from homeassistant.setup import setup_component
+from homeassistant.setup import async_setup_component
+
+from tests.common import load_fixture
+
+config = {
+    CONF_API_KEY: "foo",
+    SCAN_INTERVAL: timedelta(seconds=120),
+    CONF_TRAVEL_TIMES: [{CONF_ID: 96, CONF_NAME: "I90 EB"}],
+}
+
 
-from tests.common import get_test_home_assistant, load_fixture
+async def test_setup_with_config(hass):
+    """Test the platform setup with configuration."""
+    assert await async_setup_component(hass, "sensor", {"wsdot": config})
 
 
-class TestWSDOT(unittest.TestCase):
-    """Test the WSDOT platform."""
+async def test_setup(hass, requests_mock):
+    """Test for operational WSDOT sensor with proper attributes."""
+    entities = []
 
-    def add_entities(self, new_entities, update_before_add=False):
+    def add_entities(new_entities, update_before_add=False):
         """Mock add entities."""
         if update_before_add:
             for entity in new_entities:
                 entity.update()
 
         for entity in new_entities:
-            self.entities.append(entity)
-
-    def setUp(self):
-        """Initialize values for this testcase class."""
-        self.hass = get_test_home_assistant()
-        self.config = {
-            CONF_API_KEY: "foo",
-            SCAN_INTERVAL: timedelta(seconds=120),
-            CONF_TRAVEL_TIMES: [{CONF_ID: 96, CONF_NAME: "I90 EB"}],
-        }
-        self.entities = []
-        self.addCleanup(self.tear_down_cleanup)
-
-    def tear_down_cleanup(self):
-        """Stop everything that was started."""
-        self.hass.stop()
-
-    def test_setup_with_config(self):
-        """Test the platform setup with configuration."""
-        assert setup_component(self.hass, "sensor", {"wsdot": self.config})
-
-    @requests_mock.Mocker()
-    def test_setup(self, mock_req):
-        """Test for operational WSDOT sensor with proper attributes."""
-        uri = re.compile(RESOURCE + "*")
-        mock_req.get(uri, text=load_fixture("wsdot.json"))
-        wsdot.setup_platform(self.hass, self.config, self.add_entities)
-        assert len(self.entities) == 1
-        sensor = self.entities[0]
-        assert sensor.name == "I90 EB"
-        assert sensor.state == 11
-        assert (
-            sensor.device_state_attributes[ATTR_DESCRIPTION]
-            == "Downtown Seattle to Downtown Bellevue via I-90"
-        )
-        assert sensor.device_state_attributes[ATTR_TIME_UPDATED] == datetime(
-            2017, 1, 21, 15, 10, tzinfo=timezone(timedelta(hours=-8))
-        )
+            entities.append(entity)
+
+    uri = re.compile(RESOURCE + "*")
+    requests_mock.get(uri, text=load_fixture("wsdot.json"))
+    wsdot.setup_platform(hass, config, add_entities)
+    assert len(entities) == 1
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6333224)